io/src/main/scala/fs2/io/tcp/SocketGroup.scala

```diff
@@ -212,7 +212,8 @@ final class SocketGroup(channelGroup: AsynchronousChannelGroup, blocker: Blocker
   private def apply[F[_]](
       ch: AsynchronousSocketChannel
   )(implicit F: Concurrent[F], cs: ContextShift[F]): Resource[F, Socket[F]] = {
-    val socket = Semaphore[F](1).flatMap { readSemaphore =>
+    val socket = (Semaphore[F](1), Semaphore[F](1)).tupled.flatMap { semaphores =>
```

Comment:
Personally I'd use a pattern match like `flatMap { case (readSemaphore, writeSemaphore) =>`.

Reply:
I did that at first - the only reason I changed it to this was because the extra indentation blew up the diff, but happy to change it

Reply:
Fair enough :) let's leave it be